qbit-http-storages/src/commonMain/kotlin/qbit.storage/YandexDiskStorage.kt

```diff
@@ -0,0 +1,250 @@
+            }
+            client.close()
+            true
+        } catch (e: ClientRequestException) {
+            client.close()
+            false
+        }
+    }
+
+    private suspend fun uploadFile(client: HttpClient, urlToUploadFile: String, byteArray: ByteArray): HttpResponse {
+        return client.put(urlToUploadFile) {
+            body = ByteArrayContent(byteArray)
+        };
+    }
+
+    private suspend fun getUrlToUploadFile(client: HttpClient, filePath: String, fileName: String): HttpResponse {
+        return client.get(YANDEX_DISK_API_GET_FILE_URL) {
+            parameter("path", filePath + fileName)
+            parameter("overwrite", true)
+            header("Authorization", "OAuth $accessToken")
+        }
+    }
+
+    private suspend fun createFolderStructure(client: HttpClient, paths: List<String>) {
+        val iterator = paths.iterator()
+        while (iterator.hasNext()) {
+            val path = iterator.next()
+            client.put<String>(YANDEX_DISK_API_RESOURCES) {
+                parameter("path", path)
+                header("Authorization", "OAuth $accessToken")
+            }
+        }
+    }
+
+    private suspend fun getFile(client: HttpClient, downloadUrl: String): ByteArray {
+        val httpResponse = client.get<HttpResponse>(downloadUrl)
+        return httpResponse.readBytes()
+    }
+
+    private suspend fun getUrlToDownload(client: HttpClient, path: String): HttpResponse {
+        return client.get(YANDEX_DISK_API_GET_DOWNLOAD_FILE_URL) {
+            parameter("path", path)
+            header("Authorization", "OAuth $accessToken")
+        };
+    }
+
+    /**    Helper method for wrapping dirNames to Namespace Collection */
+    private fun wrapDirNamesToNamespaces(dirNames: List<String>, namespace: Namespace): Collection<Namespace> {
+        return dirNames.map { dirName -> Namespace(namespace, dirName) }
+    }
+
+    /**    Helper method for wrapping fileNames to Keys Collection */
+    private fun wrapFileNamesToKeys(fileNames: List<String>, namespace: Namespace): Collection<Key> {
+        return fileNames.map { filename -> Key(namespace, filename) }
+    }
+
+    /**    Helper method for getting file names by type */
+    private fun getFileNamesInResourceByType(resource: Resource, type: String): List<String> {
+        val items = resource._embedded?.items
+        var files: List<String> = emptyList()
+        if (items != null) {
+            files = items.filter { item -> item.type == type }.map { item -> item.name }
+        }
+        return files
+    }
+
+    /**    Helper method for getting full path from namespace to root */
+    private fun getFullPath(namespace: Namespace): String {
+        var parts = namespace.parts
+        return parts.joinToString("/") + "/"
+    }
+
+    /**    Helper method for getting paths from namespace and its parents */
+    private fun generatePutResourcePaths(namespace: Namespace): List<String> {
```

Comment:
Cover this method with unit test
I believe, that it will not work for composite name spaces, e.g. ns/sub1/sub2/key